src/iterable/flattenDeep.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+/**
+ * Yields a flatten array where the deep level is specified.
+ * flattenDeep(1) will have the same behavior as flatten().
+ *
+ * @this {Iterable<T>}
+ * @param depth sets the maximum depth that the generator will iterate
+ * @example Basic Usage
+ *
+ * ```javascript
+ * [[[1],[2]]]::flattenDeep(2) // yields [1,2]
+ * ```
+*/
+
+export function * flattenDeep (depth : number ) :
+  Iterable<T> {
+    if ( depth < 0 ) {
+        yield this;
+        return;
+    }
+    for ( const item of this ) {
+        if ( item[Symbol.iterator] ) {
```

Comment:
Please remove the if statement, it is unnecessary - we don't want to support `[1, [2]]::flattenDeep(100)` returning usable results.